packages/docusaurus-theme-classic/src/theme/BlogPostItem/Header/Author/index.tsx

```diff
@@ -5,11 +5,70 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import type {SVGProps} from 'react';
 import React from 'react';
 import clsx from 'clsx';
 import Link, {type Props as LinkProps} from '@docusaurus/Link';
 
 import type {Props} from '@theme/BlogPostItem/Header/Author';
+import styles from './styles.module.css';
+
+type SocialProps = SVGProps<SVGSVGElement> & {
+  user: string;
+};
+
+function getUrl(user: string, baseUrl: string) {
+  return user.startsWith('https') ? user : `${baseUrl}${user}`;
+}
```

Comment:
yaml/front matter validation/normalization should make this un-necessary

At the component level, we should always receive full URLs and not have to handle that complexity